File: Leetcode/Python/_49.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        letter_dict = dict()

        for word in strs:
            key = sorted(word)
            key = "".join(key)
            if letter_dict.get(key, None) is None:
                letter_dict[key] = []
            letter_dict[key].append(word)

        return [word for word in letter_dict.values()]

File: Leetcode/cpp/49.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::vector<std::vector<std::string>> output;
        
        std::unordered_map<std::string, std::vector<std::string>> umap;

        for (int i = 0; i < strs.size(); ++i) {
            std::string sortedWord = strs[i];
            std::sort(sortedWord.begin(), sortedWord.end());
            umap[sortedWord].push_back(strs[i]);
        }

        for (const auto& [k, v] : umap) {
            output.push_back(v);
        }

        return output;
    }
};
